src/tree/localProject/ConfigGroupTreeItem.ts

```diff
@@ -3,37 +3,101 @@
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
 
+import { readdir, readFile } from "fs-extra";
+import { basename, join } from "path";
+import { ThemeIcon } from "vscode";
+import { ext } from "vscode-azureappservice/out/src/extensionVariables";
 import { AzExtParentTreeItem, AzExtTreeItem, TreeItemIconPath } from "vscode-azureextensionui";
+import { parse } from "yaml";
 import { localize } from "../../utils/localize";
-import { treeUtils } from "../../utils/treeUtils";
-import { ConfigTreeItem } from "./ConfigTreeItem";
-import { LocalProjectTreeItem } from "./LocalProjectTreeItem";
+import { EnvironmentTreeItem } from "../EnvironmentTreeItem";
+import { GitHubConfigTreeItem } from "./ConfigTreeItem";
 
-export class ConfigGroupTreeItem extends AzExtParentTreeItem {
-    public static contextValue: string = 'azureStaticConfigGroup';
-    public contextValue: string = ConfigGroupTreeItem.contextValue;
-    public readonly label: string = localize('config', 'Configuration');
-    public parent: LocalProjectTreeItem;
+export type BuildConfig = 'app_location' | 'api_location' | 'app_artifact_location' | 'output_location';
 
-    public constructor(parent: LocalProjectTreeItem) {
+export class GitHubConfigGroupTreeItem extends AzExtParentTreeItem {
+    public static contextValue: string = 'azureStaticGitHubConfigGroup';
+    public contextValue: string = GitHubConfigGroupTreeItem.contextValue;
+    public readonly label: string = localize('gitHubConfig', 'GitHub Configuration');
+    public parent: EnvironmentTreeItem;
+    public yamlFilePath: string;
+    public buildConfigs: Map<BuildConfig, string>;
+
+    public constructor(parent: EnvironmentTreeItem, yamlFilePath: string, buildConfigs: Map<BuildConfig, string>) {
         super(parent);
+        this.yamlFilePath = yamlFilePath;
+        this.buildConfigs = buildConfigs;
+        this.id = `${parent.id}-${this.yamlFilePath}`;
+    }
+
+    public static async createGitHubConfigGroupTreeItems(parent: EnvironmentTreeItem): Promise<GitHubConfigGroupTreeItem[]> {
+        if (parent.localProjectPath && parent.inWorkspace) {
+            const treeItems: GitHubConfigGroupTreeItem[] = [];
+            const workflowsDir: string = join(parent.localProjectPath, '.github/workflows');
+            const yamlFiles: string[] = await readdir(workflowsDir);
+
+            for (const yamlFile of yamlFiles) {
+                if (yamlFile.endsWith('.yml')) {
+                    const yamlFilePath: string = join(workflowsDir, yamlFile);
```

Comment:
Can we extract all of this into its own yaml-utils file or something of that nature?  I don't want this to clutter up all of the `ConfigGroupTreeItem` code, plus yaml parsing could probably be used in other places in the future.

Reply:
I moved the majority of this code to `yamlUtils.ts` and split it up into separate functions as some other comments suggested. It did look pretty cluttered hanging out in `ConfigGroupTreeItem` 😅